warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java

```diff
@@ -0,0 +1,112 @@
+//You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+//
+
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptMapperFunction;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+import io.warp10.script.WarpScriptException;
+
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+* Mapper which returns the absolute value of the value passed as parameter
+*/
+public class MapperParseDouble extends NamedWarpScriptFunction implements WarpScriptMapperFunction {
+
+private final NumberFormat format;
+  
+  public MapperParseDouble(String name) {
+    super(name);
+    this.format = null;
+  }
+  
+  public MapperParseDouble(String name, Object language) throws WarpScriptException {   
+    super(name);
+    
+    if (language instanceof String) {
+      Locale locale = Locale.forLanguageTag((String) language);
+      format = NumberFormat.getInstance(locale);
+    } else {
+      throw new WarpScriptException("Invalid value type for " + getName() + ", expects a String");
+    }
+  }
+  
+  /**
+   * Builder for case user specify a language tag according to an
+   * IETF BCP 47 language tag string of the Locale Java class
+   */
+  public static class Builder extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+    
+    public Builder(String name) {
+      super(name);
+    }
+    
+    @Override
+    public Object apply(WarpScriptStack stack) throws WarpScriptException {
+      Object value = stack.pop();
+      
+      if (!(value instanceof String)) {
+        throw new WarpScriptException("Invalid parameter for " + getName());
+      }
+      
+      stack.push(new MapperParseDouble(getName(), value));
+      return stack;
+    }
+  }
+  
+  @Override
+  public Object apply(Object[] args) throws WarpScriptException {
+    long tick = (long) args[0];
+    String[] names = (String[]) args[1];
+    Map<String,String>[] labels = (Map<String,String>[]) args[2];
+    long[] ticks = (long[]) args[3];
+    long[] locations = (long[]) args[4];
+    long[] elevations = (long[]) args[5];
+    Object[] values = (Object[]) args[6];
+
+    if (1 != values.length) {
+      throw new WarpScriptException(getName() + " can only be applied to a single value.");
+    }
+    
+    Object value = null;
+    long location = locations[0];
+    long elevation = elevations[0];
+    
+    if (null != values[0]) {
+      if (values[0] instanceof String) {
+        try {
+          if (null == this.format) {
```

Comment:
If this case possible? Does 'value instanceof String' return true when value is null?